### PR TITLE
Enable community stake - UI & components

### PR DESCRIPTION
**Pull request by masvelio**
## Link to Issue
Closes: #5905 
Closes: #6262

## Description of Changes
- added feature flag  `FLAG_COMMUNITY_STAKE`
- added new loading spinner component `CWCircleSpinner`
- created view for third step `Community Stake` in create community flow
- created two different pages for third step and components that will be used on those pages
- adjust layout to mobile 

## Test Plan
- proper testing should be done in next step when frontend will be integrated with backend

https://github.com/hicommonwealth/commonwealth/assets/14819225/25c84064-2e1b-4f65-9f63-33fe2241288a

## Deployment Plan

- feature hidden for now behind feature flag `FLAG_COMMUNITY_STAKE=true`

### CODE REVIEW

File: packages/commonwealth/client/scripts/helpers/feature-flags.ts [hunk 1]
```diff
@@ -3,4 +3,5 @@ export const featureFlags = {
   communityHomepage: process.env.FLAG_COMMUNITY_HOMEPAGE === 'true',
   sidebarToggle: process.env.FLAG_SIDEBAR_TOGGLE === 'true',
   newAdminOnboardingEnabled: process.env.FLAG_NEW_ADMIN_ONBOARDING === 'true',
+  communityStake: process.env.FLAG_COMMUNITY_STAKE === 'true',
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/helpers/feature-flags.ts:6]: I add a placeholder for this ->  https://github.com/hicommonwealth/commonwealth/pull/6388#discussion_r1456583961. Will update once this PR is merged.

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_button.tsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ export type ButtonProps = {
   type?: 'reset' | 'submit' | 'button';
   containerClassName?: string;
 } & ButtonStyleProps &
-  React.HTMLAttributes<HTMLButtonElement>;
+  React.HTMLAttributes<HTMLButtonElement> & { form?: string };
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_button.tsx:35]: Do we need this?

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/components/ActionSteps/ActionStep.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+export interface ActionStepProps {
+  state: 'not-started' | 'completed' | 'loading';
+  label: string;
+  index: number;
+  actionButton?: {
+    label: string;
+    disabled: boolean;
+    onClick: () => void;
+  };
+}
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CreateCommunity/components/ActionSteps/ActionStep.tsx:19]: maybe we could move this to a `types.ts` file in the same folder?

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/CommunityStakeStep.scss
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/CommunityStakeStep.scss:1]: not sure if we need this in upcoming PRs?

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/CommunityStakeStep.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+import EnableStake from './EnableStake';
+import SignStakeTransactions from './SignStakeTransactions';
+
+import './CommunityStakeStep.scss';
+
+interface CommunityStakeStepProps {
+  onOptOutEnablingStake: () => void;
+  createdCommunityName: string;
+}
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/CommunityStakeStep.tsx:11]: would be cool if we could use a `types.ts` file in same folder

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/EnableStake/EnableStake.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        <CWText type="b1" className="description">
+          Community stake allows your community to fundraise via member
+          contributions. Community members can make financial contributions in
+          exchange for more voting power within the community. The more stake a
+          member has, the stronger their vote becomes. The funds are stored in a
+          secure community wallet on-chain and can be redistributed if members
+          decide to burn their stake.
+        </CWText>
+
+        {renderHint('mobile')}
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/EnableStake/EnableStake.tsx:73]: nice

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/EnableStake/EnableStake.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+import './EnableStake.scss';
+
+const validationSchema = z.object({
+  namespace: z.string().min(1, 'Namespace is required'),
+  symbol: z
+    .string()
+    .min(1, 'Symbol is required')
+    .max(9, 'Symbol should be maximum 9 characters'),
+});
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/EnableStake/EnableStake.tsx:19]: suggestion: we move this to a `validations.ts` file in same folder

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/SignStakeTransactions/SignStakeTransactions.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  onOptOutEnablingStake,
+}: SignStakeTransactionsProps) => {
+  const renderHint = (className: string) => (
+    <CreateCommunityHint
+      className={className}
+      title="What is a namespace?"
+      hint="The namespace is an address that represents your community  on-chain.
+          You can purchase additional community namespaces from the admin panel."
+    />
+  );
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/SignStakeTransactions/SignStakeTransactions.tsx:24]: i saw this used above, maybe we can make it a common component/util?

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/SignStakeTransactions/SignStakeTransactions.tsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+import { CWText } from 'views/components/component_kit/cw_text';
+import { CWButton } from 'views/components/component_kit/new_designs/cw_button';
+import ActionSteps from '../../../components/ActionSteps';
+import CreateCommunityHint from '../../../components/CreateCommunityHint';
+
+import './SignStakeTransactions.scss';
+
+interface SignStakeTransactionsProps {
+  onOptOutEnablingStake: () => void;
+}
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/SignStakeTransactions/SignStakeTransactions.tsx:13]: same `types.ts` suggestion

**Review by mzparacha — COMMENTED** [review]
Awesome work. 🔥

Codewise LGTM, testing locally.

**Review by mzparacha — APPROVED** [review]
Tested locally, works smoothly. Great work @masvelio! 🎉

File: packages/commonwealth/client/scripts/helpers/feature-flags.ts (hunk 1 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/helpers/feature-flags.ts:6]: thanks!

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/SignStakeTransactions/SignStakeTransactions.tsx (hunk 8 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/SignStakeTransactions/SignStakeTransactions.tsx:14]: per our conversation, we should create `types.ts` file if there is more than one type for file

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/SignStakeTransactions/SignStakeTransactions.tsx (hunk 7 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/SignStakeTransactions/SignStakeTransactions.tsx]: created new component

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/EnableStake/EnableStake.tsx (hunk 6 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/EnableStake/EnableStake.tsx]: moved

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/CommunityStakeStep.tsx (hunk 4 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/CommunityStakeStep.tsx:9]: same: per our conversation, we should create types.ts file if there is more than one type for file

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/CommunityStakeStep.scss
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/CommunityStakeStep.scss:1]: deleted

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/components/ActionSteps/ActionStep.tsx (hunk 3 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/CreateCommunity/components/ActionSteps/ActionStep.tsx]: moved

**Review by lzach83 — APPROVED** [review]
Tested and code looks great!

File: packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/CommunityStakeStep.tsx (hunk 4 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/CommunityStakeStep/CommunityStakeStep.tsx:9]: yup, no need to move this to types, resolving

**Pull request merged**